### PR TITLE
Fixes The Bug In Radix Tree. Issue #11316  

### LINKED ISSUE #11316: data_structures/trie/radix_tree.py wont really end up in 'case 1' for insert
### Repository commit

2e405f3

### Python version (python --version)

Python 3.11.6

### Dependencies version (pip freeze)

n/a

### Expected behavior

```
root_node = RadixNode()
root_node.insert('fooaaa')
root_node.insert('foobbb')
root_node.insert('foo')
```
Should work., i.e should have a structure like this
```
root_node.print_tree()
- foo (leaf)
-- aaa   (leaf)
-- bbb   (leaf)
```

### Actual behavior

```
root_node.insert("foo")
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[5], line 1
----> 1 root_node.insert("foo")

File ~/git/python/radix/radix.py:85, in RadixNode.insert(self, word)
     82 # Case 3: The node prefix is equal to the matching
     83 # Solution: We insert remaining word on the next node
     84 if remaining_prefix == "":
---> 85     self.nodes[matching_string[0]].insert(remaining_word)
     86 # Case 5: The word is greater equal to the matching
     87 # Solution: Create a node in between both nodes, change
     88 # prefixes and add the new node for the remaining word
     89 else:
     90     incoming_node.prefix = remaining_prefix

File ~/git/python/radix/radix.py:73, in RadixNode.insert(self, word)
     68     self.is_leaf = True
     70 # Case 2: The node has no edges that have a prefix to the word
     71 # Solution: We create an edge from the current node to a new one
     72 # containing the word
---> 73 elif word[0] not in self.nodes:
     74     self.nodes[word[0]] = RadixNode(prefix=word, is_leaf=True)
     76 else:

IndexError: string index out of range

```

### PR DESCRIPTION
**Pull request by covid-in**
### Describe your change:
Fixes  #11316 .
There is a bug in the code which doesn't handle the edge case when the insertion of empty string is done.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by imSanko — APPROVED** [review]

**Review by rtron-24 — APPROVED** [review]
I approve the changes.

**Review by hammad7 — APPROVED** [review]
Reviewed

File: data_structures/trie/radix_tree.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import unittest
```
Comment (cclauss) [inline data_structures/trie/radix_tree.py:1]: We (and the majority of the Python community) use pytest instead of unittest so all this boilerplate is not needed.

Can you please create a ___separate___ simple pull request that adds a test that proves that `word = ""` fails?  Once that is proven, we can determine how to proceed with this pull request.

File: data_structures/trie/radix_tree.py [hunk 2]
```diff
@@ -191,39 +198,42 @@ def print_tree(self, height: int = 0) -> None:
...
-    root.delete("banana")
-    assert not root.find("banana")
-    assert root.find("bananas")
+## write unit test for the code using unittest library with
+## logic similar to test_trie() function
+## and call it from main()
 
-    return True
 
+class TestRadixNode(unittest.TestCase):
```
Comment (cclauss) [inline data_structures/trie/radix_tree.py:206]: This boilerplate adds unnecessary complexity.

File: data_structures/trie/radix_tree.py (hunk 1 above)
Comment (rtron-24) [reply data_structures/trie/radix_tree.py:1]: It fails on subseunt recursive calls not on direct addition of "". Please check the issue it is trying to fix. I have added the same word insertions in Unit Test as well

File: data_structures/trie/radix_tree.py (hunk 2 above)
Comment (rtron-24) [reply data_structures/trie/radix_tree.py:206]: Sure Will Remove it. But its the best way to do this

**Review by cclauss — APPROVED** [review]
